### PR TITLE
Deal with room changes

### LINKED ISSUE #21: Handle room changes
Report by @christianschneider89 in issue #12:

> When I change the room in the SONOS App in an alarm which does already exists, HA does not recognize it.
Only after restarting the server a new entity is created with the same ID and the _2 suffix:
102596614-b022f180-4119-11eb-95eb-6c0c91f35e29

### PR DESCRIPTION
**Pull request by AaronDavidSchneider**
This issue is supposed to tackle the room current roomchange issue as outlined in #21.

To achieve this, I had to:
- rework the device structure
- clean up the code

Closes #21

### CODE REVIEW

**Comment by AaronDavidSchneider** [comment]
I would love to hear your feedback @rasmusbe and @christianschneider89!

**Comment by christianschneider89** [comment]
Works for me! Thanks!

**Comment by rasmusbe** [comment]
Looks good! Great job!

**Comment by rasmusbe** [comment]
One more question I noticed now when I trying it:
Right now the alarm gets the entity id:
`switch.sonos_alarm_1_2` where 1 is speaker id 1 (Bedroom) and 2 is alarm Id.

Is it possible to set the entity id to `switch.sonos_alarm_bedroom_2` instead or will that mess up the room changes?

**Comment by AaronDavidSchneider** [comment]
Thanks for getting back!

> Right now the alarm gets the entity id:
> `switch.sonos_alarm_1_2` where 1 is speaker id 1 (Bedroom) and 2 is alarm Id.

Could you please test if that persists when you readd the integration? It sounds to me that this is due to the reworking of the devices.

> Is it possible to set the entity id to `switch.sonos_alarm_bedroom_2` instead or will that mess up the room changes?

The idea right now is that we use the alarm ids as only basis for the `unique_id`. Since that one doesn't change upon room changes. We can therefore easily track the alarm itself as it changes room names, etc.

The entity id is therefore in fact only a composition of the `alarm_id`
```python
self._id = alarm._alarm_id
_entity_id = slugify("sonos_alarm_{}".format(self._id))
```

TLDR:
I think this update of the component requires the user to readd the integration.

**Comment by AaronDavidSchneider** [comment]
Did it work upon readding the itegration @rasmusbe? (see my comment above)
As soon as it does, I will release it

**Comment by rasmusbe** [comment]
I will try it tomorrow, unable to try it today unfortunately

**Comment by rasmusbe** [comment]
Re-adding did solve it, I made some changes in #25 to, in my opinion, improve the device info a bit.
